docs/development.md

```diff
@@ -1,78 +1,73 @@
 # Development
 
-## golangci-lint
+## Table Of Contents
 
-A [golangci-lint](https://golangci-lint.run/) [config](../.golangci.yaml) and [Makefile](../Makefile) target are provided to keep the codebase aligned to certain best practices and styles.
+- [Running Locally](#running-locally)
+- [Testing](#testing)
+- [API Changes](#api-changes)
+- [Development Hints](#development-hints)
+- [Controllers](#controllers)
+- [Golang CI Linter](#golang-ci-linter)
+- [Pre-commit Hooks](#pre-commit-hooks)
 
-The target installs golangci-lint if it is not already present. To run use the following command:
-
-```shell
-make lint
-```
-
-The target is also used when running unit tests locally.
-
-## pre-commit
-
-A [pre-commit](https://pre-commit.com/) [config](../.pre-commit-config.yaml) is provided to help developers catch any simple mistakes prior to committing their changes.
-
-To install and use the tool please run the following commands:
+## <a name="running-locally"></a> Running Locally
 
-```shell
-pip install --user pre-commit
-pre-commit install
-git commit -s
-```
-
-## Running locally
-
-The operator can run locally on the developer's machine.
-It will watch the cluster configured in a file pointed to by the `$KUBECONFIG`.
+The operator can run locally on the developer's machine,
+monitoring the cluster specified in a file indicated by the `$KUBECONFIG`
+environment variable:
```

Comment:
```suggestion
The operator can run locally on the developer's machine.
It will monitor the cluster specified in a file indicated by the `$KUBECONFIG`
environment variable:
```